alpine/FreeElectronLaser.cpp

```diff
@@ -0,0 +1,1851 @@
+        t[2] = bunchInit.sigmaGammaBeta_[2] * sqrt( - 2.0 * log( generate(6, i + Np0) ) ) * cos( 2.0 * M_PI * generate(7, i + Np0) );
+
+        if ( fabs(r[0]) < bunchInit.tranTrun_ && fabs(r[1]) < bunchInit.tranTrun_ && fabs(r[2]) < bunchInit.longTrun_)
+          {
+            /* Shift the generated charge to the center position and momentum space.			*/
+            //charge.rnp    = bunchInit.position_[ia];
+            charge.rnp    = bunchInit.position_;
+            charge.rnp   += r;
+
+            charge.gb   = gb;
+            charge.gb  += t;
+            //std::cout << gb << "\n";
+            if(std::isinf(gb[2])){
+                std::cerr << "it klonked here\n";
+            }
+
+            /* Insert this charge and the mirrored ones into the charge vector.				*/
+            insertCharge(charge);
+          }
+      }
+
+    /* If the longitudinal type of the bunch is uniform a tapered part needs to be added to remove the
+     * CSE from the tail of the bunch.									*/
+    if ( bunchInit.distribution_ == "uniform" ){
+      for ( ; i < unsigned( uint32_t(Np / ng) * ( 1.0 + 2.0 * bunchInit.lambda_ * sqrt( 2.0 * M_PI ) / ( 2.0 * bunchInit.sigmaPosition_[2] ) ) ); i += size)
+        {
+            
+          r[0] = bunchInit.sigmaPosition_[0] * sqrt( - 2.0 * log( generate(0, i + Np0) ) ) * cos( 2.0 * M_PI * generate(1, i + Np0) );
+          r[1] = bunchInit.sigmaPosition_[1] * sqrt( - 2.0 * log( generate(0, i + Np0) ) ) * sin( 2.0 * M_PI * generate(1, i + Np0) );
+
+          /* Determine the longitudinal coordinate.							*/
+          r[2] = 2.0 * bunchInit.lambda_ * sqrt( - 2.0 * log( generate(2, i + Np0) ) ) * sin( 2.0 * M_PI * generate(3, i + Np0) );
+          r[2] += ( r[2] < 0.0 ) ? ( - bunchInit.sigmaPosition_[2] ) : ( bunchInit.sigmaPosition_[2] );
+
+          /* Determine the transverse momentum.								*/
+          t[0] = bunchInit.sigmaGammaBeta_[0] * sqrt( - 2.0 * log( generate(4, i + Np0) ) ) * cos( 2.0 * M_PI * generate(5, i + Np0) );
+          t[1] = bunchInit.sigmaGammaBeta_[1] * sqrt( - 2.0 * log( generate(4, i + Np0) ) ) * sin( 2.0 * M_PI * generate(5, i + Np0) );
+          t[2] = bunchInit.sigmaGammaBeta_[2] * sqrt( - 2.0 * log( generate(6, i + Np0) ) ) * cos( 2.0 * M_PI * generate(7, i + Np0) );
+          //std::cerr << "DOING UNIFORM tapering!!!\n";
+          if ( fabs(r[0]) < bunchInit.tranTrun_ && fabs(r[1]) < bunchInit.tranTrun_ && fabs(r[2]) < bunchInit.longTrun_)
+            {
+                //std::cerr << "ACTUALLY DOING UNIFORM tapering!!!\n";
+              /* Shift the generated charge to the center position and momentum space.			*/
+              charge.rnp   = bunchInit.position_[ia];
+              charge.rnp  += r;
+
+              charge.gb  = gb;
+              
+              charge.gb += t;
+              //std::cout << gb[0] << "\n";
+              //if(std::isinf(gb.squaredNorm())){
+              //    std::cerr << "it klonked here\n";
+              //}
```

Comment:
there are a lot of commented prints which seem to be leftovers from debugging...